lib/ansible/plugins/inventory/nmap.py

```diff
@@ -127,6 +127,7 @@ def parse(self, inventory, loader, path, cache=False):
             ip = None
             ports = []
             for line in stdout.splitlines():
+                line = to_text(line)
```

Comment:
* For performance, this should be performed on stdout prior to splitting it into separate strings.
* For robustness, it should use errors='surrogate_or_strict' to throw a UnicodeError in case of non-utf-8 input rather than mangling (if you choose to handle it, probably deciding to ignore the whole line rather than just individual bytes in each line.)